src/game/option_graphics.c

```diff
@@ -195,64 +262,68 @@ void Option_Graphics(INVENTORY_ITEM *inv_item)
         switch (g_OptionSelected) {
         case TEXT_PERSPECTIVE:
             if (g_Config.rendering.enable_perspective_filter) {
-                g_Config.rendering.enable_perspective_filter = 0;
-                reset = true;
+                g_Config.rendering.enable_perspective_filter = false;
+                reset = TEXT_PERSPECTIVE;
             }
             break;
 
         case TEXT_BILINEAR:
             if (g_Config.rendering.enable_bilinear_filter) {
-                g_Config.rendering.enable_bilinear_filter = 0;
-                reset = true;
+                g_Config.rendering.enable_bilinear_filter = false;
+                reset = TEXT_BILINEAR;
             }
             break;
 
         case TEXT_VSYNC:
             if (g_Config.rendering.enable_vsync) {
-                g_Config.rendering.enable_vsync = 0;
-                reset = true;
+                g_Config.rendering.enable_vsync = false;
+                reset = TEXT_VSYNC;
                 GFX_Context_SetVSync(g_Config.rendering.enable_vsync);
             }
             break;
 
         case TEXT_BRIGHTNESS:
             if (g_Config.brightness > MIN_BRIGHTNESS) {
                 g_Config.brightness -= 0.1f;
-                reset = 1;
+                reset = TEXT_BRIGHTNESS;
             }
             break;
 
         case TEXT_UI_TEXT_SCALE:
             if (g_Config.ui.text_scale > MIN_UI_SCALE) {
                 g_Config.ui.text_scale -= 0.1;
-                reset = true;
+                reset = TEXT_UI_TEXT_SCALE;
             }
             break;
 
         case TEXT_UI_BAR_SCALE:
             if (g_Config.ui.bar_scale > MIN_UI_SCALE) {
                 g_Config.ui.bar_scale -= 0.1;
-                reset = true;
+                reset = TEXT_UI_BAR_SCALE;
             }
             break;
 
         case TEXT_RESOLUTION:
             if (Screen_SetPrevRes()) {
-                reset = true;
+                reset = TEXT_RESOLUTION;
             }
             break;
         }
     }
 
     if (g_InputDB.deselect || g_InputDB.select) {
-        reset = true;
+        reset = TEXT_NUMBER_OF;
     }
 
-    if (reset) {
+    if (reset == TEXT_NUMBER_OF) {
         for (int i = 0; i < TEXT_NUMBER_OF; i++) {
             Text_Remove(m_Text[i]);
             m_Text[i] = NULL;
         }
+        m_IsTextInit = false;
+        Settings_Write();
+    } else if (reset > -1) {
+        Option_GraphicsChangeTextOption(reset);
```

Comment:
Are we sure we don't need to `Settings_Write` otherwise?

Reply:
It writes the settings in both the if and else if.

Reply:
No because it's the case where no option was selected. We don't need to save settings every loop right? Maybe we need to make it save in the initial case?